obvs/lenses.py

```diff
@@ -314,47 +312,55 @@ class PatchscopeLogitLens(BaseLogitLens):
 
     The source layer l and position i can vary.
 
-    In words: The logit-lens maps the hidden state at position i of layer l of the model M
-        to the last layer of that same model. It is equal to taking the hidden state and
-        applying unembed to it.
+        In words: The logit-lens maps the hidden state at position i of layer l of the model M
+            to the last layer of that same model. It is equal to taking the hidden state and
+            applying unembed to it.
     """
 
-    def run(self, substring: str, layers: list[int]):
-        """Run the logit lens for each layer in layers and each token in substring.
-
-        Args:
-            substring (str): Substring of the prompt for which the top prediction and logits
-                should be calculated.
-            layers (List[int]): Indices of Transformer Layers for which the lens should be applied
+    def __init__(self, model: str, prompt: str, device: str, layers: list[int], substring: str):
+        """
+        substring (str): Substring of the prompt for which the top prediction and logits should be calculated.
+        layers (list[int]): Indices of Transformer Layers for which the lens should be applied
         """
 
-        # get starting position and tokens of substring
+        super().__init__(model, prompt, device)
         start_pos, substring_tokens = self.patchscope.source_position_tokens(substring)
-
-        # initialize tensor for logits
+        self.start_pos = start_pos
+        self.layers = layers
```

Comment:
We could maybe move these attributes into the BaseLogitLens class and also use them in the ClassicLogitLens

Reply:
Agreed, thanks! Made this change